perf/perf_pcp.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2019 IBM.
+# Author: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import distro, process, cpu
+from avocado.utils.software_manager import SoftwareManager
+
+
+class PCP(Test):
+
+    """
+    Performance Co Pilot (PCP)
+    1. Starts pmcd daemon
+    2. Install perfevent pmda
+    3. Test pcp command
+    4. Test 24x7 events
+    5. Remove perfevent pmda
+    """
+
+    def setUp(self):
+        smg = SoftwareManager()
+        self.cpu_arch = cpu.get_cpu_arch().lower()
+        dist = distro.detect()
+        if dist.name in ['centos', 'fedora', 'rhel', 'SuSE']:
+            pkgs = ['pcp', 'pcp-pmda-perfevent']
+        else:
+            self.cancel("PCP is not supported on %s" % dist.name)
+
+        for pkg in pkgs:
+            if not smg.check_installed(pkg) and not smg.install(pkg):
+                self.cancel("Package %s is missing/could not be installed"
+                            % pkg)
+
+    def lpar_24x7_ppc64le_check(self):
+        if 'ppc' not in process.system_output("uname -p", ignore_status=True):
+            self.cancel("Processor is not ppc64")
+        # Check if this is a guest
+        # 24x7 is not suported on guest
+        cpu_output = process.run("cat /proc/cpuinfo")
+        if "emulated by" in cpu_output.stdout:
+            self.cancel("24x7: This test is not supported on guest")
+
+        # Check if 24x7 is present
+        if not os.path.exists("/sys/bus/event_source/devices/hv_24x7"):
+            self.cancel("%s doesn't exist.This feature is supported"
+                        " only on LPAR" % self.event_sysfs)
+
+    def test_pmcd_daemon(self):
+        output = process.run("systemctl start pmcd", shell=True)
+        output = process.run("systemctl status pmcd", shell=True)
+        if "active (running)" not in output.stdout:
+            self.fail("PCP: Can not start pmcd daemon")
+
+    def test_pmda_perfevent_install(self):
+        if not os.path.isfile("/var/lib/pcp/pmdas/perfevent/Install"):
+            self.fail("PCP: perfevent pmda package not installed")
+        cmd = "cd /var/lib/pcp/pmdas/perfevent/; echo pipe | ./Install"
```

Comment:
Better to use `os.chdir()` read file with python and pass to `./Install` script.

Reply:
@harish-24 when `os.chdir()` used need to add `try ... catch` code. `process.run` code is better than `adding os.chdir and process.run`

Reply:
@nasastry As this is a avocado/python based framework it is better to have maximum possible code in python to reduce execution time and moreover builtin functions are already available. Also adding `try...catch` does not add any overhead.